src/test/java/com/google/devtools/build/lib/actions/ResourceManagerTest.java

```diff
@@ -248,11 +287,26 @@ public void testThatTestsCannotBeOverallocated() throws Exception {
     assertThat(e).hasCauseThat().hasMessageThat().contains("is still alive");
   }
 
+  @Test
+  public void testThatExtraResourcesCannotBeOverallocated() throws Exception {
+    assertThat(rm.inUse()).isFalse();
+
+    // Given a partially acquired extra resources:
+    acquire(0, 0, ImmutableMap.of("gpu", 1.0f), 1);
+
+    // When a request for extra resources is made that would overallocate,
+    // Then the request fails:
+    TestThread thread1 = new TestThread(() -> acquire(0, 0, ImmutableMap.of("gpu", 1.1f), 0));
+    thread1.start();
+    AssertionError e = assertThrows(AssertionError.class, () -> thread1.joinAndAssertState(10000));
```

Comment:
10 s for one test is a lot. I suppose we could wait 1 sec in thread and then dlip flag to understand the current state